### PR TITLE
feat: git integration

### PR DESCRIPTION
**Pull request by pheel**
This adds the capability to sync project state with a git repo.

### CODE REVIEW

File: botfront/imports/ui/components/utils/TrainButton.jsx [hunk 1]
```diff
@@ -3,106 +3,340 @@ import PropTypes from 'prop-types';
 import { Meteor } from 'meteor/meteor';
```
Comment (GuillaumeTech) [inline botfront/imports/ui/components/utils/TrainButton.jsx:3]: This file could be splited into two compoments one for training and one for git

**Review by GuillaumeTech — CHANGES_REQUESTED** [review]
after a revert  the availables intent are not updated, in this clip I revert to a commit where there was no nlu data but i can still see the intents, it's solved when I reload the page
https://user-images.githubusercontent.com/27060431/103658271-fc9f6700-4f38-11eb-9787-a8822acb6ac1.mov

- It would be nice to have a loader on the git button so after a commit and push the user know that something is happening
-  It would be nice to have a or separator between the https and ssh part as only one of those is required 
image
- I think only one line break after the "or" in this tooltip would make it easier to read 
image

File: botfront/imports/lib/importers/validateDomain.js [hunk 2]
```diff
@@ -204,10 +204,12 @@ const validateADomain = (
...
-    // do not import slots that are in current default domain or are programmatically generated
-    [...Object.keys(defaultSlots), ...INTERNAL_SLOTS].forEach((k) => {
-        delete slotsFromFile[k];
-    });
+    if (!isDefaultDomain) {
+        // do not import slots that are in current default domain or are programmatically generated
+        [...Object.keys(defaultSlots), ...INTERNAL_SLOTS].forEach((k) => {
+            delete slotsFromFile[k];
+        });
+    }
```
[the remark this answers is not in the document]
Comment (pheel) [reply botfront/imports/lib/importers/validateDomain.js:212]: 👍

File: botfront/imports/api/graphql/project/import.utils.js [hunk 3]
```diff
@@ -201,10 +193,7 @@ export async function importSteps({
         summary: wipeProject ? [{ text: 'ALL PROJECT DATA WILL BE ERASED.' }] : [],
     };
     const filesAndValidationData = await readAndValidate(await unzipFiles(files), params);
-
-    if (onlyValidate || hasErrors(filesAndValidationData.fileMessages)) {
-        return filesAndValidationData;
-    }
+    if (onlyValidate) return filesAndValidationData;
```
[the remark this answers is not in the document]
Comment (pheel) [reply botfront/imports/api/graphql/project/import.utils.js:196]: The error I was getting was empty training data file. I thought having these errors made sense in the UI as a way to exclude empty files, and are a little more elegant than just secretly doing nothing with them. I guess you had a different vision of errors where they would require user action in every case.

**Review by GuillaumeTech — APPROVED** [review]

File: botfront/cypress/package.json [hunk 4]
```diff
@@ -10,16 +10,16 @@
...
-    "cypress": "3.6.0",
-    "cypress-plugin-retries": "1.2.2",
-    "js-yaml": "^3.13.1"
-  },
-  "dependencies": {
     "glob": "^7.1.6",
+    "js-yaml": "^3.13.1",
     "jszip": "^3.5.0",
-    "path": "^0.12.7"
+    "keypair": "^1.0.1",
```
Comment (GuillaumeTech) [inline botfront/cypress/package.json:20]: This lib has a GPL license 😢  iirc we would have to distribute botfront under GPL to be compliant ? so we can't use this lib @znat ?

**Review by GuillaumeTech — CHANGES_REQUESTED** [review]
checkout the comment about gpl

**Review by GuillaumeTech — APPROVED** [review]

File: botfront/imports/api/importExport/export.methods.js [hunk 5]
```diff
@@ -32,7 +53,329 @@ if (Meteor.isServer) {
...
+        const nodegit = require('nodegit');
+        const {
+            dir, index, repo, branchCommit, exclusions,
+        } = repoInfo;
+        // given that the present method is called on training, calling 'exportRasa'
+        // here means 'rasa.getTrainingPayload' will be called twice in short succession,
+        // which is a bit stupid, but we live with it for now.
+        const zip = await Meteor.callWithPromise('exportRasa', projectId, 'all', {
+            noBlob: true,
+        });
```
Comment (GuillaumeTech) [inline botfront/imports/api/importExport/export.methods.js:203]: it does not export conversation or incoming with this call

**Review by GuillaumeTech — CHANGES_REQUESTED** [review]

File: botfront/imports/api/importExport/export.methods.js (hunk 5 above)
Comment (pheel) [reply botfront/imports/api/importExport/export.methods.js:203]: It was on purpose, since we are versioning a bot, not data from bot usage...

Comment (GuillaumeTech) [reply botfront/imports/api/importExport/export.methods.js:203]: damn I've made a commit for that

**Pull request merged**